src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,230 @@
+        Topic newAwsRegion = deploymentConfiguration.getAWSRegion();
+        return !Topic.compareValue(currentAwsRegion, newAwsRegion);
+    }
+
+    private boolean mqttHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        Topics currentMqtt = currentdeviceConfiguration.getMQTTNamespace();
+        Topics newMqtt = deploymentConfiguration.getMQTTNamespace();
+        return !Topics.compareChildren(currentMqtt, newMqtt);
+    }
+
+    private boolean iotDataEndpointHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentIotDataEndpoint = currentdeviceConfiguration.getIotDataEndpoint();
+        Topic newIotDataEndpoint = deploymentConfiguration.getIotDataEndpoint();
+        return !Topic.compareValue(currentIotDataEndpoint, newIotDataEndpoint);
+    }
+
+    private boolean greengrassDataPlaneEndpointHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentGreengrassDataPlaneEndpoint = currentdeviceConfiguration.getGGDataEndpoint();
+        Topic newGreengrassDataPlaneEndpoint = deploymentConfiguration.getGGDataEndpoint();
+        return !Topic.compareValue(currentGreengrassDataPlaneEndpoint, newGreengrassDataPlaneEndpoint);
+    }
+
+    private boolean greengrassDataPlanePortHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        Topic currentGreengrassDataPlanePort = currentdeviceConfiguration.getGreengrassDataPlanePort();
+        Topic newGreengrassDataPlanePort = deploymentConfiguration.getGreengrassDataPlanePort();
+        return !Topic.compareValue(currentGreengrassDataPlanePort, newGreengrassDataPlanePort);
+    }
+
+    /**
+     * Creates an MQTT client and checks if the device can connect to AWS.
+     *
+     * @param totallyCompleteFuture Future will be updated if validation fails
+     */
+    public boolean mqttClientCanConnect(CompletableFuture<DeploymentResult> totallyCompleteFuture) {
+        logger.atDebug().log("Checking MQTT client can connect");
+
+        String clientId = Coerce.toString(deploymentConfiguration.getThingName()) + clientIdSuffix;
+        try (AwsIotMqttConnectionBuilder builder = mqttClient.createMqttConnectionBuilder(deploymentConfiguration,
+                securityService, null);
+                MqttClientConnection connection = builder.withClientId(clientId).build()) {
+            int operationTimeoutMillis = mqttClient.getMqttOperationTimeoutMillis(deploymentConfiguration);
+            connection.connect().get(operationTimeoutMillis, TimeUnit.MILLISECONDS);
+            return true;
+        } catch (MqttException | TimeoutException | ExecutionException e) {
+            logger.atWarn().cause(e).log("mqtt Client failed to connect with the new configuration");
+        } catch (InterruptedException e) {
+            logger.atError().log("Wait for mqtt Client to connect was interrupted");
```

Comment:
why is this at error, but the log above is at warn? This does not seem any more serious. If anything it is less serious